src/wd.rs

```diff
@@ -373,3 +372,318 @@ impl NewSessionResponse {
         }
     }
 }
+
+/// The builder of [`PrintConfiguration`].
+#[derive(Debug)]
+pub struct PrintConfigurationBuilder {
+    orientation: PrintOrientation,
+    scale: f64,
+    background: PrintBackground,
+    size: PrintSize,
+    margins: PrintMargins,
+    page_ranges: Vec<PrintPageRange>,
+    shrink_to_fit: bool,
+}
+
+impl Default for PrintConfigurationBuilder {
+    fn default() -> Self {
+        Self {
+            orientation: PrintOrientation::default(),
+            scale: 1.0,
+            background: PrintBackground::default(),
+            size: PrintSize::default(),
+            margins: PrintMargins::default(),
+            page_ranges: Vec::default(),
+            shrink_to_fit: true,
+        }
+    }
+}
+
+impl PrintConfigurationBuilder {
+    /// Builds the [`PrintConfiguration`].
+    ///
+    /// Returns None if:
+    ///  - the scale, the margins or the size are infinite, NaN or negative
+    ///  - the margins overflow the size (e.g. margins.left + margins.right >= page.width)
+    pub fn build(self) -> Option<PrintConfiguration> {
+        let must_be_finite_and_positive = [
+            self.scale,
+            self.margins.top,
+            self.margins.left,
+            self.margins.right,
+            self.margins.bottom,
+            self.size.width,
+            self.size.height,
+        ];
+        if !must_be_finite_and_positive
+            .into_iter()
+            .all(|n| n.is_finite() && n.is_sign_positive())
+        {
+            return None;
+        }
+
+        if (self.margins.top + self.margins.bottom) >= self.size.height
+            || (self.margins.left + self.margins.right) >= self.size.width
+        {
+            return None;
+        }
+
+        Some(PrintConfiguration {
+            orientation: self.orientation,
+            scale: self.scale,
+            background: self.background,
+            size: self.size,
+            margins: self.margins,
+            page_ranges: self.page_ranges,
+            shrink_to_fit: self.shrink_to_fit,
+        })
+    }
+
+    /// Sets the orientation of the printed page.
+    ///
+    /// Default: [`PrintOrientation::Portrait`].
+    pub fn orientation(mut self, orientation: PrintOrientation) -> Self {
+        self.orientation = orientation;
+
+        self
+    }
+
+    /// Sets the scale of the printed page.
+    ///
+    /// Default: 1.
+    pub fn scale(mut self, scale: f64) -> Self {
+        self.scale = scale;
+
+        self
+    }
+
+    /// Sets whether or not to print the backgrounds of the page.
+    ///
+    /// Default: [`PrintBackground::Exclude`].
+    pub fn background(mut self, background: PrintBackground) -> Self {
```

Comment:
I'm okay with this one being a `bool` instead. Or are you expecting this to be a non-binary choice in the future?